packages/make-styles/src/types.ts

```diff
@@ -2,22 +2,91 @@ import * as CSS from 'csstype';
 
 export type MakeStylesCSSValue = string | 0;
 
+type MakeStylesUnsupportedCSSProperties = {
+  // We don't support expansion of CSS shorthands
+  animation?: never;
+  background?: never;
+  border?: never;
+  borderBlock?: never;
+  borderBlockEnd?: never;
+  borderBlockStart?: never;
+  borderBottom?: never;
+  borderColor?: never;
+  borderImage?: never;
+  borderInline?: never;
+  borderInlineEnd?: never;
+  borderInlineStart?: never;
+  borderLeft?: never;
+  borderRadius?: never;
+  borderRight?: never;
+  borderStyle?: never;
+  borderTop?: never;
+  borderWidth?: never;
+  columnRule?: never;
+  flex?: never;
+  flexFlow?: never;
+  font?: never;
+  gap?: never;
+  grid?: never;
+  gridArea?: never;
+  gridColumn?: never;
+  gridGap?: never;
+  gridRow?: never;
+  gridTemplate?: never;
+  listStyle?: never;
+  margin?: never;
+  mask?: never;
+  maskBorder?: never;
+  offset?: never;
+  outline?: never;
+  overflow?: never;
+  padding?: never;
+  placeItems?: never;
+  placeSelf?: never;
+  textDecoration?: never;
+  textEmphasis?: never;
+  transition?: never;
+};
 type MakeStylesCSSProperties = Omit<
   CSS.Properties<MakeStylesCSSValue>,
   // We have custom definition for "animationName"
   'animationName'
->;
+> &
+  MakeStylesUnsupportedCSSProperties;
```

Comment:
Shouldn't these go inside of the `Omit` type?

Reply:
TLDR; No.

---

As we don't have enough strict typings due selectors we allow to use eighter `MakeStylesCSSValue` (`0` or `string`) or `MakeStylesStyle` to handle following:

```tsx
{ '--var(foo)': '5px' } // is not know property, template literals as keys are supported from TS 4.4 😿

{ '.foo': { color: 'red' } } // we cannot type this selector
```

Once we will use `Omit` we will remove definitions:

```tsx
// currently
{ padding: '5px' } // throws as `padding` is `never`

// with Omit
{ padding: '5px' } // passes as assignable to `MakeStylesCSSValue`
```

----

I have stricter typings for TS 4.4, once we will ship separate typings for different TS versions I will try to make this better.